src/main/java/net/citizensnpcs/EventListen.java

```diff
@@ -559,4 +570,45 @@ public int hashCode() {
             return prime * (prime * (prime + ((worldName == null) ? 0 : worldName.hashCode())) + x) + z;
         }
     }
+
+    private class SkinUpdateTracker {
+        float initialYaw;
+        final Location location = new Location(null, 0, 0, 0);
+        boolean hasMoved;
+        int rotationCount;
+
+        SkinUpdateTracker(Player player) {
+            reset(player);
+        }
+
+        boolean shouldUpdate(Player player) {
+
+            if (!hasMoved) {
+                hasMoved = true;
+                return true;
+            }
+
+            Location currentLoc = player.getLocation(YAW_LOCATION);
+            float currentYaw = currentLoc.getYaw();
+
+            boolean hasRotated = Math.abs(currentYaw - this.initialYaw) < 90;
```

Comment:
I think Minecraft yaws need to be changed using two while loops to put them in the right range. You may end up with bugs if you use it like this directly. Also, is this just for small rotations? Not sure why 90 is being used here.

while(yaw < 180) yaw += 180;
while(yaw > 180) yaw -= 180;